### ISSUE TITLE
Database path on mobile platform doesn't _really_ work

### ISSUE THREAD

**Issue by vlussenburg** [backtrace-labs/backtrace-unity#23]
Hi,

Resources:
- [Application Sandbox on iOS](https://cocoacasts.com/what-is-application-sandboxing)
- [Unity persistDataPath](https://docs.unity3d.com/ScriptReference/Application-persistentDataPath.html?_ga=2.52937004.580650304.1593634622-1704455353.1584134619) 
- [Unity temporaryCachePath](https://docs.unity3d.com/ScriptReference/Application-temporaryCachePath.html)

1) Providing a full database path does not make sense in a Unity mobile context. Most directories are sandboxed. It provides users with an ability to shoot themselves in the foot and it's a bad practice from a Unity perspective (stay within designated path)
2) For relative paths, the wrong directory is used:

In `BacktraceConfiguration.cs#ValidateDatabasePath`
```
            if (!Path.IsPathRooted(databasePathCopy))
            {
                databasePathCopy = Path.GetFullPath(Path.Combine(Application.dataPath, databasePathCopy));
           }          
```

`Application.dataPath` is not writeable: `Contains the path to the game data folder on the target device (Read Only).`

The real kicker is that the prefix of `Application.dataPath` is only done during validating and never persisted, in other words, that validation is living in its own world :)

Suggestion: change implementation to prefix `temporaryCachePath` before any relative path provided as default functionality. This should work fine on MacOS, Windows and Android (not all tested yet) too.

**Comment by vlussenburg** [comment]
Implemented in the current (sloppy) fork:
https://github.com/vlussenburg/backtrace-unity

**Comment by vlussenburg** [comment]
- [ ] Windows Standalone
- [x] MacOS Editor
- [ ] MacOs Standalone
- [x] iOS
- [x] Android

**Comment by konraddysput** [comment]
```
Providing a full database path does not make sense in a Unity mobile context. Most directories are sandboxed. It provides users with an ability to shoot themselves in the foot and it's a bad practice from a Unity perspective (stay within designated path)
```

Backtrace integration allows to provide short version of database path - for example "./Database" should point to `./application/directory/database` directory.

```
For relative paths, the wrong directory is used:
```

`temporaryCachePath ` isn't a good option for this kind of problem. Name suggests something else - meanwhile here we want to store data that can be used after game restart/PC restart or PC cleanup.

Based on this: 
https://docs.unity3d.com/ScriptReference/Application.html
and our users feedback, our game developers is mostly using dataPath or persistendDataPath.

Right now we will try by default generate database in persistendDataPath. Please see diff here: https://github.com/backtrace-labs/backtrace-unity/pull/33

**Comment by vlussenburg** [comment]
Tested master @ 9f298881bb070a7da8ac748a0fe701b45becc4c8

- [ ]  Windows Standalone
- [ ]  MacOS Editor
- [ ]  MacOs Standalone
- [ ]  iOS
- [ ]  Android
- [ ]  WebGL

**Comment by konraddysput** [comment]
Fix will be available in version 3.0.1 (already on master)
